### DISCUSSION TITLE
feat: backlink should jump to correspondent line in file

### DISCUSSION THREAD

**Discussion by UnkwUsr** [zk-org/zk-nvim#129] [Ideas]
I see there 2 solutions:

1. Implement it on `zk` server side and then do apropriate steps on vim plugin side.
2. Implement it only on vim plugin side. (dirty patch: https://github.com/UnkwUsr/zk-nvim/commit/e10629f031dafc909ba5f494c74cd0468ca7e2c7)
3. (NEW) We can partly reimplement `:ZkBacklinks` function in our personal config, which gives us much more ways to customize behavior. This solution does not require any patches to zk. Example: https://github.com/UnkwUsr/dotfiles/commit/561715979bd1ccf9917d02e32afd77449042ee28

**Comment by mickael-menu** [comment]
I'm not sure this can be easily added to the `zk` server, as `ZkBacklinks` uses `zk.list` down the hood, which has no notion of a target column:line.

Alternatively, you can use LSP references for backlinks, which work with column:line. But of course the UI is not as nice.

```
map('n', '<leader>zb', '<Cmd>lua vim.lsp.buf.references()<CR>', opts)
```

**Reply by lazywei** [reply]
I think this is great. Combining with other UI such as telescope / lspsaga etc. makes this very easy to use and to navigate. Thanks.

**Comment by UnkwUsr** [comment]
@mickael-menu Can you reopen it please? I'm staying with my custom solution and it works flawlessly for me (idk, maybe I'm not using some features that this can break, as I see that my modified function is part of api).

I think this feature can be very useful for other users if they see this issue and give it a try, and so then things also may go forward.

Or, maybe I should open draft pull req with temporary solution?

**Comment by UnkwUsr** [comment]
One more solution I made to be able to customize behavior much more. This does not require any patches to zk, everything is done in your personal config: https://github.com/UnkwUsr/dotfiles/commit/561715979bd1ccf9917d02e32afd77449042ee28
